### ISSUE TITLE
Feature: Delayed messaging

### ISSUE THREAD

**Issue by beljebuzz** [SilenceIM/Silence#423]
Hi all, this is a feature request: Wait for N configurable seconds before sending the message in order to give the user a chance to review/cancel the action and edit the message. i.e:

1. The user configures the delay parameter to be 5 seconds (5, 6, whichever he/she prefers).
2. The user writes a message and push the send button.
3. The app turns the send button into a "cancel" button and shows a count down indicator for 5 seconds.
4. Case 1: The user taps the "cancel" button before the count down reaches zero. The app cancels the send action an allows the user to edit the message and start this process again.
4. Case 2: The count down reaches zero and the app sends the message.

This is a cool feature I miss from other SMS apps.

Thank you very much!

**Comment by Salamandar** [comment]
More largely, it would be great to let third party apps to send requests to Silence (and other sms apps). Like an intent "please send this message". With per-app authorizations inside Silence. 
That would allow AirDroid to send messages and/or access (if the user allows it) the Silence DB. And the same for Signal. Etc etc.

**Comment by rugk** [comment]
Yes @Salamander this is a good idea. I'd also appreciate it, because then I could use QkSMS.

However this is a different issue, so could you please create a new one or look for an existing one?

**Comment by Salamandar** [comment]
Okay, doing that.

**Comment by Salamandar** [comment]
Here : #487
